VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
         return result;
     }
 
+    private List<EnrolledCoursesResponse> getEnrolledCourses() {
+        List<EnrolledCoursesResponse> ret = null;
+
+        try {
+            ret = environment.getServiceManager().getEnrolledCourses(false);
```

Comment:
Yeah, a challenge that I ran into while doing this was just figuring out
what APIs I should use and why. If we are going to cache, we need to
incorporate the storage API. I don’t know if anyone else has comments about
the high level class relationships. This was the best that I could figure
out how to do without more guidance.

On Wed, May 18, 2016 at 10:59 AM, Christopher Lee <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java
> <https://github.com/edx/edx-app-android/pull/707#discussion_r63719592>:
>
> > @@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
> >          return result;
> >      }
> >
> > +    private List<EnrolledCoursesResponse> getEnrolledCourses() {
> > +        List<EnrolledCoursesResponse> ret = null;
> > +
> > +        try {
> > +            ret = environment.getServiceManager().getEnrolledCourses(false);
>
> By reverting to the older API, I believe we will not be able to the
> refresh the access_token on expiration.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/707/files/0c45f11a71cef83eb283660b17c3042aa536934f#r63719592>
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX